nemo/collections/asr/metrics/audio.py

```diff
@@ -0,0 +1,199 @@
+        """Update the underlying metric by taking into account channel selector and input length.
+
+        Args:
+            preds: tensor with predictions, shape (B, C, T)
+            target: tensor with target signals, shape (B, C, T)
+            input_length: Optional, input tensor with length (in samples) of each signal in the batch, shape (B,).
+                          If not provided, it is assumed that all samples are valid.
+
+        Returns:
+            Underlying metric averaged on the current batch.
+        """
+        preds, target = self._select_channel(preds=preds, target=target)
+
+        if input_length is None:
+            self._metric.update(preds=preds, target=target)
+        else:
+            # Each example in this batch has a different length
+            for b_preds, b_target in self._trim_inputs(preds=preds, target=target, input_length=input_length):
+                self._metric.update(preds=b_preds, target=b_target)
+
+    def compute(self) -> torch.Tensor:
+        """Compute the underlying metric.
+        """
+        return self._metric.compute()
+
+    @torch.jit.unused
+    def forward(
+        self, preds: torch.Tensor, target: torch.Tensor, input_length: Optional[torch.Tensor] = None
+    ) -> torch.Tensor:
+        """Call underlying forward method to add the batch statistics to the accumulated metric state
+        and return the result for the current batch.
+
+        Args:
+            preds: tensor with predictions, shape (B, C, T)
+            target: tensor with target signals, shape (B, C, T)
+            input_length: Optional, input tensor with length (in samples) of each signal in the batch, shape (B,).
+                          If not provided, it is assumed that all samples are valid.
+
+        Returns:
+            Underlying metric averaged on the current batch.
+        """
+        preds, target = self._select_channel(preds=preds, target=target)
+
+        if input_length is None:
+            return self._metric(preds=preds, target=target)
+        else:
+            # Each example in this batch has a different length
+            batch_values = []
+            for b_preds, b_target in self._trim_inputs(preds=preds, target=target, input_length=input_length):
+                batch_values.append(self._metric(preds=b_preds, target=b_target))
+            # Average over the batch
+            return self._batch_reduction(batch_values)
+
+    def reset(self) -> None:
+        """Reset the underlying metric.
+        """
+        self._metric.reset()
+
+    def __repr__(self) -> str:
+        """Return string representation of the object.
+        """
+        _op_metric = f"(metric: {repr(self._metric)}, channel: {self._channel})"
+        repr_str = self.__class__.__name__ + _op_metric
+
+        return repr_str
+
+    def _wrap_compute(self, compute: Callable) -> Callable:
```

Comment:
Using private methods like this is going to break down the road cause PTL changes all of these internals very frequently